src/coreclr/jit/morph.cpp

```diff
@@ -14290,6 +14290,45 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
             }
             break;
 
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
+        case GT_HWINTRINSIC:
+        {
+            GenTreeHWIntrinsic* hw = tree->AsHWIntrinsic();
+            switch (hw->gtHWIntrinsicId)
+            {
+                case NI_SSE2_Xor:
+                case NI_AVX2_Xor:
+                {
```

Comment:
This should either handle SSE_Xor & AVX_Xor or it should explictly check for and skip floating-point (`SSE2_Xor` includes `double`)